linfa-ensemble/src/random_forest/algorithm.rs

```diff
@@ -0,0 +1,191 @@
+use crate::random_forest::hyperparameters::RandomForestParams;
+use linfa_predictor::{LinfaError, Predictor, ProbabilisticPredictor};
+use linfa_trees::DecisionTree;
+use ndarray::Array;
+use ndarray::Axis;
+use ndarray::{Array1, ArrayBase, Data, Ix1, Ix2};
+use ndarray_rand::rand_distr::Uniform;
+use ndarray_rand::RandomExt;
+use std::collections::HashMap;
+
+/// A random forest is composed of independent decision trees performing a prediction and collecting each of them
+pub struct RandomForest {
+    pub hyperparameters: RandomForestParams,
+    pub trees: Vec<DecisionTree>,
+}
+
+impl Predictor for RandomForest {
+    /// Return predicted class for each sample calculated with majority voting
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - A 2D array of floating point elements
+    ///
+    ///
+    fn predict(
+        &self,
+        x: &ArrayBase<impl Data<Elem = f64>, Ix2>,
+    ) -> Result<Array1<u64>, LinfaError> {
+        let ntrees = self.hyperparameters.n_estimators;
+        assert!(ntrees > 0, "Run .fit() method first");
```

Comment:
I think this has been discussed in the comments already but just a note here to replace with something like LinfaError instead of a panic.